File: clients/javascript/src/models/msgQueueAckIn.ts
// this file is @generated

export interface MsgQueueAckIn {
    namespace?: string | null;
    msgIds: string[];
}

export interface MsgQueueAckIn_ {
    namespace?: string | null;
    topic: string;
    consumerGroup: string;
    msgIds: string[];
}

export const MsgQueueAckInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): MsgQueueAckIn_ {
        return {
            namespace: object['namespace'],
            topic: object['topic'],
            consumerGroup: object['consumer_group'],
            msgIds: object['msg_ids'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: MsgQueueAckIn_): any {
        return {
            'namespace': self.namespace,
            'topic': self.topic,
            'consumer_group': self.consumerGroup,
            'msg_ids': self.msgIds,
        };
    }
}

File: z-clients/python/diom/models/msg_queue_ack_in.py
# this file is @generated
import typing as t

from ..internal.base_model import BaseModel


class MsgQueueAckIn(BaseModel):
    namespace: str | None = None

    msg_ids: t.List[str]


class _MsgQueueAckIn(BaseModel):
    namespace: str | None = None

    topic: str

    consumer_group: str

    msg_ids: t.List[str]
